src/test/java/com/ays/assignment/service/impl/AssignmentServiceImplTest.java

```diff
@@ -93,4 +103,91 @@ void givenAssignmentId_whenAssignmentNotFound_thenThrowAysAssignmentNotExistById
         Mockito.verify(identity, Mockito.times(1))
                 .getInstitutionId();
     }
+
+    @Test
+    void givenAssignmentListRequest_whenAssignmentStatusIsAvailable_thenReturnAysPageAssignmentResponse() {
+
+        // Given
+        String mockInstitutionId = AysRandomUtil.generateUUID();
+
+        AssignmentListRequest assignmentListRequest = new AssignmentListRequestBuilder().withValidValuesForStatus().build();
+
+        List<AssignmentEntity> mockAssignmentEntities = Collections.singletonList(new AssignmentEntityBuilder()
+                .withValidFields().build());
+        Page<AssignmentEntity> mockPageAssignmentEntities = new PageImpl<>(mockAssignmentEntities);
+
+        List<Assignment> mockAssignments = ASSIGNMENT_ENTITY_TO_ASSIGNMENT_MAPPER.map(mockAssignmentEntities);
+        AysPage<Assignment> mockAysPageAssignments = AysPage.of(assignmentListRequest.getFilter(), mockPageAssignmentEntities, mockAssignments);
+
+        // When
+        Mockito.when(identity.getInstitutionId()).thenReturn(mockInstitutionId);
+        Mockito.when(assignmentRepository.findAll(Mockito.any(Specification.class), Mockito.any(Pageable.class)))
+                .thenReturn(mockPageAssignmentEntities);
+
+        // Then
+        AysPage<Assignment> aysPageAssignment = assignmentService.getAssignments(assignmentListRequest);
+
+        AysPageBuilder.assertEquals(mockAysPageAssignments, aysPageAssignment);
+
+        Mockito.verify(assignmentRepository, Mockito.times(1))
+                .findAll(Mockito.any(Specification.class), Mockito.any(Pageable.class));
```

Comment:
Burada `identity.getInstitutionId()` metodunu da verify etmemiz en doğrusu olur çünkü oradaki metoda gidip gitmediğini sorgulamamız gerekiyor

Reply:
Verify etmeyi denemiştim. Fakat tahmin ediyorum ki `identity.getInstitutionId()` bir lambda fonksiyonunun içinde olduğu için verify edemiyor.

```java
    @Override
    public AysPage<Assignment> getAssignments(AssignmentListRequest listRequest) {

        Specification<AssignmentEntity> byStatusAndPhoneNumber = listRequest.toSpecification(AssignmentEntity.class);
        Specification<AssignmentEntity> byInstitutionId = (root, query, criteriaBuilder) ->
                criteriaBuilder.equal(root.get("institutionId"),identity.getInstitutionId());

        Page<AssignmentEntity> assignmentEntities = assignmentRepository
                .findAll(byStatusAndPhoneNumber.and(byInstitutionId),listRequest.toPageable());

        List<Assignment> assignments = assignmentEntityToAssignmentMapper.map(assignmentEntities.getContent());

        return AysPage.of(
                listRequest.getFilter(),
                assignmentEntities,
                assignments
        );
    }
```
Şu şekilde dışarı alırsam verify edebiliyorum. Fakat bu da çirkin ve bi tık gereksiz bir kod. Bu değişikliği yapayım mı? @agitrubard 

```java
Long institutionId = identity.getInstitutionId();
```

Reply:
Hallettim. @agitrubard